File: core/src/main/java/io/snabble/sdk/BarcodeFormat.java
package io.snabble.sdk;

public enum BarcodeFormat {
    CODE_39,
    CODE_128,
    EAN_8,
    EAN_13,
    ITF,
    QR_CODE,
    DATA_MATRIX;

    public static BarcodeFormat parse(String str) {
        switch (str) {
            case "code39":
                return CODE_39;
            case "code128":
                return CODE_128;
            case "ean8":
                return EAN_8;
            case "ean13":
                return EAN_13;
            case "itf14":
                return ITF;
            case "datamatrix":
                return DATA_MATRIX;
            case "qr":
                return QR_CODE;
        }

        return null;
    }
}

File: core/src/main/java/io/snabble/sdk/BarcodeFormat.kt
package io.snabble.sdk

/**
 * Enum describing a barcode format
 */
enum class BarcodeFormat {
    CODE_39,
    CODE_128,
    EAN_8,
    EAN_13,
    ITF_14,
    QR_CODE,
    DATA_MATRIX,
    PDF_417;

    companion object {
        /**
         * Parse the BarcodeFormat from a string
         */
        fun parse(str: String?): BarcodeFormat? {
            return when (str) {
                "code39" -> CODE_39
                "code128" -> CODE_128
                "ean8" -> EAN_8
                "ean13" -> EAN_13
                "itf14" -> ITF_14
                "datamatrix" -> DATA_MATRIX
                "pdf417" -> PDF_417
                "qr", "qrCode" -> QR_CODE
                else -> null
            }
        }
    }
}